plugins/module_utils/network/sonic/config/ssh/ssh.py

```diff
@@ -0,0 +1,352 @@
+
+        changed_ssh_facts = self.get_ssh_facts()
+
+        result['before'] = existing_ssh_facts
+        if result['changed']:
+            result['after'] = changed_ssh_facts
+
+        new_config = changed_ssh_facts
+        old_config = existing_ssh_facts
+        if self._module.check_mode:
+            result.pop('after', None)
+            new_config = get_new_config(commands, existing_ssh_facts)
+            result['after(generated)'] = new_config
+        if self._module._diff:
+            result['diff'] = get_formatted_config_diff(old_config, new_config, self._module._verbosity)
+
+        result['warnings'] = warnings
+        return result
+
+    def set_config(self, existing_ssh_facts):
+        """ Collect the configuration from the args passed to the module,
+            collect the current configuration (as a dict from facts)
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        want = self._module.params['config']
+        have = existing_ssh_facts
+        resp = self.set_state(want, have)
+        return to_list(resp)
+
+    def set_state(self, want, have):
+        """ Select the appropriate function based on the state provided
+
+        :param want: the desired configuration as a dictionary
+        :param have: the current configuration as a dictionary
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        state = self._module.params['state']
+        if state == 'overridden':
+            commands, requests = self._state_replaced_overridden(want, have)
+        elif state == 'deleted':
+            commands, requests = self._state_deleted(want, have)
+        elif state == 'merged':
+            commands, requests = self._state_merged(want, have)
+        elif state == 'replaced':
+            commands, requests = self._state_replaced_overridden(want, have)
+        return commands, requests
+
+    def _state_replaced_overridden(self, want, have):
+        """ The command generator when state is replaced or overridden
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        commands = []
+        requests = []
+        state = self._module.params['state']
+
+        client_want = {}
+        client_have = {}
+        if want and want.get('client', None):
+            client_want['client'] = want['client']
+        if have and have.get('client', None):
+            client_have['client'] = have['client']
+        client_commands, client_requests = self.handle_ssh_client_replaced_overridden(client_want, client_have)
+        requests.extend(client_requests)
+
+        if client_commands and len(requests) > 0:
+            commands = update_states(client_commands, state)
```

Comment:
Corresponding to the changes requested in handle_ssh_client_replaced_overridden to avoid overwriting "deleted" commands, please replace these lines with something like:


```suggestion
        commands, requests = self.handle_ssh_client_replaced_overridden(client_want, client_have)
```

Reply:
Addressed.